source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -756,27 +756,30 @@ def get_journey_extremity_pt_section(journey, criteria):
     return extremity_pt_section
 
 
-def get_journey_pt_extremity(journey, criteria):
-    extremity_pt_section = get_journey_extremity_pt_section(journey, criteria)
+def get_journey_pt_extremity(journey, attractivities_virtual_fallbacks):
+    extremity_pt_section = get_journey_extremity_pt_section(journey, attractivities_virtual_fallbacks)
 
     if extremity_pt_section is None:
         return None, None
 
-    if criteria == "arrival_stop_attractivity":
+    if attractivities_virtual_fallbacks.get("arrival"):
         return extremity_pt_section, extremity_pt_section.destination
     else:
         return extremity_pt_section, extremity_pt_section.origin
 
 
-def compute_journey_virtual_duration_and_attractivity(journey, criteria, virtual_fallbacks, attractivities):
-    extremity_pt_section, extremity = get_journey_pt_extremity(journey, criteria)
+def compute_journey_virtual_duration_and_attractivity(journey, attractivities_virtual_fallbacks):
+    extremity_pt_section, extremity = get_journey_pt_extremity(journey, attractivities_virtual_fallbacks)
     if extremity_pt_section is None:
         return journey.duration, 0
+    if attractivities_virtual_fallbacks.get("arrival"):
+        attractivity_virtual_fallback = attractivities_virtual_fallbacks.get("arrival", {}).get(extremity.uri)
+    else:
+        attractivity_virtual_fallback = attractivities_virtual_fallbacks.get("departure", {}).get(extremity.uri)
+    virtual_fallback = attractivity_virtual_fallback.virtual_duration or 0
```

Comment:
does this crash when `attractivity_virtual_fallback` is `None` because `extremity.uri` was not found in `attractivities_virtual_fallbacks` ?

Reply:
Done